Realm/RLMSyncSubscription.mm

```diff
@@ -18,38 +18,144 @@
 
 #import "RLMSyncSubscription.h"
 
+#import "RLMObjectSchema_Private.hpp"
+#import "RLMObject_Private.hpp"
+#import "RLMProperty_Private.hpp"
 #import "RLMRealm_Private.hpp"
 #import "RLMResults_Private.hpp"
 #import "RLMUtil.hpp"
 
+#import "object_store.hpp"
 #import "sync/partial_sync.hpp"
 
 using namespace realm;
 
 @interface RLMSyncSubscription ()
-- (instancetype)initWithName:(NSString *)name results:(Results const&)results realm:(RLMRealm *)realm;
-
 @property (nonatomic, readwrite) RLMSyncSubscriptionState state;
 @property (nonatomic, readwrite, nullable) NSError *error;
 @end
 
 @implementation RLMSyncSubscription {
+    @public
+    NSString *_name;
+}
+
+- (instancetype)initPrivate {
+    return (self = [super init]);
+}
+
+- (void)unsubscribe {
+    __builtin_unreachable();
+}
+@end
+
+@interface RLMSyncSubscriptionObject : RLMObjectBase
+@end
+@implementation RLMSyncSubscriptionObject {
+    util::Optional<NotificationToken> _token;
+    Object _obj;
+}
+
+- (NSString *)name {
+    return _row.is_attached() ? RLMStringDataToNSString(_row.get_string(_row.get_column_index("name"))) : nil;
```

Comment:
Are managed RealmObjects normally allowed to be accessed when the Realm is closed? In Java this would throw an exception.

Reply:
No, this class is specifically trying to act like the pre-existing RLMSyncSubscription type, which isn't a managed object.